### ISSUE TITLE
[wishlist] Snap symbolic icon

### ISSUE THREAD

**Issue by fthx** [ubuntu/yaru#2403]
Hi,

I was writing a snap manager extension for Gnome and I noticed that there's no snap symbolic icon. Could it be created even if it does not exist in Adwaita?

**Comment by fthx** [comment]
Ok, I created a transitional one:
![image](https://user-images.githubusercontent.com/23138504/95677133-bb125080-0bc3-11eb-9c6d-e733fba3e37c.png)

I do what I can. I don't know how to make an SVG. And Inkscape snap isn't working. :-p

**Comment by fthx** [comment]
Looks better in panel:
![image](https://user-images.githubusercontent.com/23138504/95677164-f280fd00-0bc3-11eb-8e92-f2136cfccd83.png)

**Comment by Feichtmeier** [comment]
Awesome. I think we can add it. Could you make a PR? Thanks

**Comment by fthx** [comment]
Sorry for my limited knowledge but I don't know how to to that.
What are the steps to achieve that? I tried but without success... "I do what I can." :-)

**Comment by Muqtxdir** [comment]
hi @fthx,

sorry for being this late, how does this look?


![snap-symbolic](https://user-images.githubusercontent.com/54065734/95677433-2ddef400-0be3-11eb-85c6-8ad216bc70c5.png)


zip file containing the svg : [snap-symbolic.zip](https://github.com/ubuntu/yaru/files/5360991/snap-symbolic.zip)

**Comment by fthx** [comment]
Looks great to me:
![image](https://user-images.githubusercontent.com/23138504/95677591-d3379f00-0bc6-11eb-87a5-0bca2e3903f4.png)
Better than mine, for sure! Happy to see a faster-than-light icon creation!

**Comment by fthx** [comment]
You can find the new extension version including your icon here:
https://extensions.gnome.org/review/download/19853.shell-extension.zip

**Comment by fthx** [comment]
I'm here for another issue but I see this one still open.
The icon from @Muqtxdir is very good, is it now included in Yaru icon theme?

**Comment by fthx** [comment]
Bump?

**Comment by Muqtxdir** [comment]
> The icon from @Muqtxdir is very good, is it now included in Yaru icon theme?

hey @fthx sorry for the late reply, could you please tell what should be the name given to the snap symbolic?

I was under the assumption that the snap symbolic was only for your extension, but it seems this needs to be included in yaru as well (am I right?)

**Comment by fthx** [comment]
Hi,

I don't manage the Yaru project so it's not my turn to respond!
:-) 
IMHO including this icon in Yaru makes sense since Yaru is Ubuntu's default theme and Snaps are important things nowadays.

**Comment by Jupiter007-43** [comment]
If everyone is OK to include it of course (but I don't see why a new icon could hurt 😉), simply naming it snap-symbolic seems IMO a good idea.

**Comment by Feichtmeier** [comment]
Yes do it 👍

**Comment by elsiehupp** [comment]
I know I'm a bit late to the game, but I noticed that the version of the `snap-symbolic` icon wasn't really optimized for itty bitty sizes, so I put together a couple of subtly tweaked variations.

From left, these are:
* The version posted here
* A [pixel-hinted](https://danellebailey.com/design-blog/2016/9/29/pixel-hinting-for-crisp-icons) version
* A pixel-hinted version with blunt tips

The icons in these embeds should appear as 16x16 on a 100% screen and as 32x32 on a 200% screen (with 48x48 or 96x96 black backgrounds for preview purposes):

<img width="48px" alt="snap-symbolic-exact" src="https://gist.githubusercontent.com/elsiehupp/0b03cdc7a4fcd4010e4f26a663ce8974/raw/b565135659fac78ce4e6b093757c76d20b9d17e8/snap-symbolic-exact.svg"> <img width="48px" alt="snap-symbolic-hinted" src="https://gist.githubusercontent.com/elsiehupp/b6efdc0e674e0e0d6a06962eebe4a444/raw/12127f674c8a7c5bb564651b12d0546f5a5c55cc/snap-symbolic-hinted.svg"> <img width="48px" alt="snap-symbolic-blunt" src="https://gist.githubusercontent.com/elsiehupp/08c6276192c2920e38bb4eaceac4dd88/raw/b6c00c0fe9bce22b0f6c07ba36ba859a0452b807/snap-symbolic-blunt.svg">

I would say the plain pixel-hinted version looks "sharp", while the blunted version is probably the most legible without squinting (especially on a 100% screen).

I would recommend temporarily changing the screen zoom ratio in order to see how the icons render at different sizes.

It would also probably help to look at the icons in context, so here are actually usable versions:
* [snap-symbolic-exact.svg](https://gist.githubusercontent.com/elsiehupp/7d71466dbed2193c10417e4922b4c7af/raw/da94642475ef1c524064bb1af2679b6bc969c293/snap-symbolic-exact.svg)
* [snap-symbolic-hinted.svg](https://gist.githubusercontent.com/elsiehupp/852f704bf5949dc422809fa759da382f/raw/ab6a8a1acf9725e487376039b31fcd3a60777ade/snap-symbolic-hinted.svg)
* [snap-symbolic-blunt.svg](https://gist.githubusercontent.com/elsiehupp/5d7c6b622217101e1ecdf442bd2720bb/raw/3263cbf772c38fc4a7aac5405fdb324bd80e1c6c/snap-symbolic-blunt.svg)

I guess the point I'm trying to make is that the highest fidelity may or may not actually be the clearest version of an icon at extremely small sizes.

Feel free to do what you want with the linked gists, though I would be happy to do a pull request if you'd like.

**Comment by fthx** [comment]
Hi,

Personally, I don't like the third, just because that's not the original snap logo. I agree it looks nice, though!

So I tried the second one in Snap Manager extension context. Here is what is rendered in top panel (so, small icons) on my 14" FHD display:
![Capture d’écran de 2021-05-13 10-50-59](https://user-images.githubusercontent.com/23138504/118103345-187cb300-b3da-11eb-8668-819cdc071bcb.png)
I must confess I prefer the @Muqtxdir one, but yours is black, so that's not fair on a rather blackish top panel. Could you post your hinted one without black color, like @Muqtxdir one?

**Comment by elsiehupp** [comment]
For the hinted one, I actually did fit it to a 14x14 frame, because the norm seems to be for there to be a 1px margin. I could make a full 16x16 version, though.

As far as the color goes, I actually stripped out all color data from the SVG code. If it’s not the right color, it’s probably because GTK isn’t recognizing it as a symbolic icon, which might be due to the fact that the `-symbolic` suffix isn’t actually at the end of the filename.

Here are links to the SVG code (rather than links to the files themselves, though you can still view the previews):
* `snap-symbolic-exact.svg` \[[code](https://gist.github.com/elsiehupp/7d71466dbed2193c10417e4922b4c7af?short_path=43a2df9)\] \[[rendered](https://gist.github.com/elsiehupp/7d71466dbed2193c10417e4922b4c7af)\]
* `snap-symbolic-hinted.svg` \[[code](https://gist.github.com/elsiehupp/852f704bf5949dc422809fa759da382f?short_path=1d6f9ed)\] \[[rendered](https://gist.github.com/elsiehupp/852f704bf5949dc422809fa759da382f)\]
* `snap-symbolic-blunt.svg` \[[code](https://gist.github.com/elsiehupp/5d7c6b622217101e1ecdf442bd2720bb?short_path=cac8eef)\] \[[rendered](https://gist.github.com/elsiehupp/5d7c6b622217101e1ecdf442bd2720bb)\]

I stripped out all of the unnecessary bits from the SVG—using a text editor—so it’s actually almost human-readable!

**Comment by elsiehupp** [comment]
Here’s a a version that’s hinted and 16x16:

<img alt="snap-symbolic-hinted-16x16" width="48px" src="https://gist.githubusercontent.com/elsiehupp/8b2433a2f99478cbd24bbc4fe6ae56e4/raw/c7ed0ab7f5f6e4ac8458d2a9f020184fc6de6a65/snap-symbolic-hinted-16x16-preview.svg">

\[[code](https://gist.github.com/elsiehupp/7651c9d709842d2e57c0e3118033134d?short_path=64e57b4)\] \[[rendered](https://gist.github.com/elsiehupp/7651c9d709842d2e57c0e3118033134d)\]

**Comment by elsiehupp** [comment]
Again, probably the way to get GTK to recolor the SVG files is to change the filenames so they’re something like `snap-hinted-16x16-symbolic.svg` (e.g. ending in `-symbolic`) rather than `snap-symbolic-hinted-16x16.svg`. But that’s just a guess.

**Comment by elsiehupp** [comment]
What I mean by “pixel hinting” is a lot clearer if you look at the two versions of the icon with an overlaid pixel grid:

#### Original
<img width="424" alt="snap-symbolic-exact-grid" src="https://user-images.githubusercontent.com/9206310/118109487-4698cd80-b3af-11eb-93b0-c190cf1a6fc1.png">

#### Pixel-Hinted

<img width="425" alt="snap-symbolic-hinted-grid" src="https://user-images.githubusercontent.com/9206310/118109497-4b5d8180-b3af-11eb-9dd3-6712aad29093.png">

Basically the difference is that the “pixel-hinted” version of the icon sticks to exact pixel increments as much as possible. This is mainly noticeable with making the negative-space lines exactly one pixel wide, but I lined everything else up with the pixel grid as much as possible regardless. (FWIW the 1px gap between the body and the head required some pencil-and-paper trigonometry, because my drawing program isn’t CAD…)

**Comment by fthx** [comment]
[-symbolic end did the trick.]

So:
[14" FHD display in top panel]

Old one:
![image](https://user-images.githubusercontent.com/23138504/118109573-aad48500-b3e1-11eb-8d05-23fb9d83a44e.png)

Hinted one:
![image](https://user-images.githubusercontent.com/23138504/118109677-ca6bad80-b3e1-11eb-9d60-fbb4162519ed.png)

The new one seems crispier, it looks better IMO. It looks good too on my 24" FHD monitor.
And it looks definitely better using your 16² setup.

**Comment by elsiehupp** [comment]
![SOOOO CRISPY!!](https://gist.githubusercontent.com/elsiehupp/852652b5483a23ddefa57dc72c1b5009/raw/fff411ef5da3f2a4c1b1292da3c1abb6729c7604/crispy.gif)

**Comment by fthx** [comment]
Ok I updated.

**Comment by Feichtmeier** [comment]
@elsiehupp thank you very much. Yes we welcome pull requests 🙂

**Comment by Muqtxdir** [comment]
Hi @elsiehupp 👋

The new icons that you've suggested are soooo much better than mine and I didn't manage to put a pull requests for the `snap-symbolic` in the first place, so as @Feichtmeier you can push a pull request for this `snap-symbolic` to be added in yaru
